aws/associate-jc-system.ps1

```diff
@@ -32,18 +32,43 @@ Function AssociateJcSystem() {
   Add-JCSystemGroupMember -SystemID $JcSystemId -ByID -GroupID $JcSystemsGroupId
 }
 
+# Get the User ID that match an email address from a list of JC users
+# synapse.org emails are stored in a JC custom user attribute
+Function Get-JCUserId {
+    Param($Users, $Email)
+    if ($Email.Contains("synapse.org")) {
+        foreach ($User in $Users) {
+            foreach ($Attribute in $User.attributes) {
+                if ($Attribute.name -match "SynapseEmail" -and $Attribute.value -match $Email) {
+                    $UserId = $User._id
+                }
+            }
+        }
+    } else {
+        foreach ($User in $Users) {
+            if ($User.email -eq $Email) {
+                $UserId = $User._id
+            }
+        }
+    }
+    return $UserId
```

Comment:
This is a fine solution. IIUC Jumpcloud doesn't allow users to set custom attributes, so might be hard to maintain. Should this function handle errors, such what would happen if someone creates a Windows instances, but they do not have a Jumpcloud account? I think the instance would eventually boot, but just wouldn't be accessible (in a lot of ways), so maybe its not important to handle that error.